ext/standard/user_filters.stub.php

```diff
@@ -5,7 +5,7 @@
 class php_user_filter
 {
     /** @return int */
-    public function filter($in, $out, &$consumed, $closing) {}
+    public function filter(mixed $in, mixed $out, mixed &$consumed, mixed $closing) {}
```

Comment:
Are those types right? Manual lists this as:
`public filter ( resource $in , resource $out , int &$consumed , bool $closing ) : int`

Reply:
It is:

```
PHP_METHOD(php_user_filter, filter)
{
	zval *in, *out, *consumed, *closing;
	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zzzz", &in, &out, &consumed, &closing) == FAILURE) {
		RETURN_THROWS();
	}
}
```

Unless I'm looking at the wrong code (which has some chance since it doesn't seem a very useful method), it is correct.

Reply:
This "class" is really more of an interface. Or maybe we can think of it as an abstract base class with dummy default implementations. We should adjust the types match what it will actually be called with if possible, to allow extended classes to use types as well.